bn-apps/cordapp-updates-distribution/README.md

```diff
@@ -0,0 +1,270 @@
+
+## Usage
+
+`corda-updates-shell` supports the following modes:
+
+* **INIT**. Initialises an empty local repository and creates a sample configuration file.
+```bash
+# Will initialize an empty repository and create a settings.yaml file under USER.HOME/.corda-updates folder
+java -jar corda-updates-shell-xxx.jar --mode=INIT
+
+# Will initialize an empty repository and create a settings.yaml file under ~/.my-repo
+java -jar corda-updates-shell-xxx.jar --mode=INIT --configPath="~/.my-repo"
+
+```  
+* **SYNC**. Synchronises the contents of the local repository with the remote repositories, configured in the `settings.yaml`. All versions missing from the local repository will be downloaded during the synchronisation. 
+```bash
+# Will pull down locally missing versions for all CorDapps configured in the settings.yaml file. 
+java -jar corda-updates-shell-xxx.jar --mode=SYNC
+
+# Will pull down locally missing versions of "net.corda:corda-finance" CorDapp starting from the version 0 and up to the version 2.0 not inclusively.
+java -jar corda-updates-shell-xxx.jar --mode=SYNC --cordapp="net.corda:corda-finance:[,2.0)"
+
+```  
+* **PRINT_VERSIONS**. Prints available versions of the specified CorDapp to the screen.
+```bash
+# Will print all available versions of "net.corda:corda-finance" CorDapp. 
+java -jar corda-updates-shell-xxx.jar --mode=PRINT_VERSIONS --cordapp="net.corda:corda-finance:[,)"
+```
+
+Configuration file is looked up in the following order:
+* A custom path if one was provided via `--configPath` parameter 
+* `settings.yaml` in the current working folder
+* `settings.yaml` in `USER.HOME/.corda-updates`
+
+## How to use the shell
+
+1. Download `corda-updates-shell` jar or build it by yourself. TODO: download link here
+2. Initialise a local repository via `java -jar corda-updates-shell-xxx.jar --mode=INIT`
+3. Add CorDapps that you would like to watch to `settings.yaml`
+4. Start using the utility by invoking `SYNC` or `PRINT_VERSIONS` commands
+
+# corda-updates-app
+
+`corda-updates-app`is a CorDapp that provides the following functionality:
+* A scheduled state to periodically check for new updates
+* Flows to schedule updates and retrieve available CorDapp versions
+* Flows to report CorDapp version information to the BNO
+
+## Scheduling synchronisation
+
+Updates can be scheduled via `ScheduleSyncFlow`. Synchronisation interval is driven by `syncInterval` configuration property and defaults to *5 hours*. `ScheduleSyncFlow` can be started from shell or RPC. 
+
+```kotlin
+// synchronisation should be launched asynchronously if corda-flows transport is used 
+subFlow(ScheduleSyncFlow(launchAsync = true))
+```
+
+> It is better to avoid launching ScheduleSyncFlow in synchronous mode, as it might result to a deadlock, as described in one of the previous sections
```

Comment:
It does work for non-flow transports. If a user knows that he is not interested in flow-based transports - then he can get results synchronously (instead of going via @CordaService).

Reply:
That wasn't clear to me from reading the docs - can it be clarified?

Reply:
Yeps, added more information on that